### PR TITLE
Weblogic empty servlet path fallback

### PR DESCRIPTION
**Pull request by SylvainJuge**
## What does this PR do?

Improves Servlet path naming on Weblogic when `use_path_as_transaction_name=true` is set.

## Checklist

- [x] This is a bugfix
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [x] I have added tests that would fail without this fix

### CODE REVIEW

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java
Comment (felixbarny) [inline apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java:245]:
```suggestion
                if ((servletPath == null || servletPath.length() == 0) && servletContext != null) {
```

IINM, it would be equivalent to `if (servletPath == null || (servletPath.length() == 0 && servletContext != null))` otherwise.

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java
Comment (felixbarny) [inline apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java:251]: Much cleaner to have every argument in a separate line. I like!

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java
Comment (felixbarny) [inline apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java:149]: We'd expect this case to be relatively rare, right? It does cause some allocations but if it's an edge case I think that's fine.

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java
Comment (SylvainJuge) [reply apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java:245]: Good catch!

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java
Comment (SylvainJuge) [reply apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java:149]: Yes, this is definitely an edge case as most of the time we have `servletPath` provided as we usually expect.

File: integration-tests/simple-webapp/src/main/webapp/WEB-INF/web.xml
Comment (felixbarny) [inline integration-tests/simple-webapp/src/main/webapp/WEB-INF/web.xml:36]: Does this reproduce the exact issue we're seeing on Weblogic?

Comment (SylvainJuge) [reply integration-tests/simple-webapp/src/main/webapp/WEB-INF/web.xml:36]: Not directly, if I understand the servlet spec correctly this should in theory make `servletPath` to be empty, but in practice with Tomcat this is not the case. for example we get `servletPath` = `/test-servlet` and `pathInfo` = `/info` when request URL is `/test-servlet/info` which is also a bit confusing. I am not sure it's really worth adding an integration test for that case though, I've kept it mostly for manual testing.

**Review by felixbarny — APPROVED** [review]
LGTM, but let's wait for the feedback before merging.

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java
Comment (felixbarny) [inline apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java:153]: nit: I'd guard this with a if so that the varargs array is only allocated when the debug log level is enabled.

**Pull request merged**
